plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationServiceInitializer.kt

```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notifications
+
+import com.intellij.ide.util.RunOnceUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.ProjectActivity
+
+internal class NotificationServiceInitializer : ProjectActivity {
+
+    override suspend fun execute(project: Project) {
+        val service = NotificationPollingService.getInstance()
+        ProcessNotificationsBase()
+        RunOnceUtil.runOnceForApp(this::class.qualifiedName.toString()) {
```

Comment:
Do we need this? Will this startup activity anyway run only once?

Reply:
Switched to Atomic Boolean, but the same sort of logic applies. Because StartupActivity is obsolete in favor of ProjectActivity, we need to ensure we are only creating one instance of the polling service per IDE session. Otherwise, when a user opens multiple projects, it would start multiple instances of the polling service